tests-arquillian/src/test/java/org/jboss/weld/tests/interceptors/producer/InterceptionFactoryTest.java

```diff
@@ -88,4 +91,22 @@ public void testMapPut(@Produced Map<String, Object> map) {
         assertEquals(1, Producer.INVOCATIONS.size());
         assertEquals(Arrays.toString(params), Producer.INVOCATIONS.get(0));
     }
+    
+    @Test(expected = IllegalStateException.class)
+    public void testListAdd(BeanManager bm) {
+        // resolving via Instance just to make the NPE exception human-readable (direct method injection will blow up with Arq. stack)
+        // Invalid producer using an InterceptionFactory for List (interface) and applying it to ArrayList
+        List<Object> list = bm.createInstance().select(new TypeLiteral<List<Object>>() {}, Produced.Literal.INSTANCE).get();
```

Comment:
Because it was the example test I had on my branch and I did it this is way to make the exception human-readable (as the comments suggest). Otherwise you got an NPE coming from arquillian.
I agree it should be either changed now, or at least the comment is now useless.